java/org/apache/catalina/realm/DataSourceRealm.java

```diff
@@ -572,6 +801,21 @@ protected void startInternal() throws LifecycleException {
         temp.append(" = ?");
         preparedCredentials = temp.toString();
 
+        // Create the user attributes PreparedStatement string (only its tail w/o SELECT
+        // clause)
+        temp = new StringBuilder(" FROM ");
+        temp.append(userTable);
+        temp.append(" WHERE ");
+        temp.append(userNameCol);
+        temp.append(" = ?");
+        preparedAttributesTail = temp.toString();
+
+        // Create the available user attributes PreparedStatement string
+        temp = new StringBuilder("SELECT * FROM ");
+        temp.append(userTable);
+        temp.append(" WHERE FALSE");
```

Comment:
Who guarantees that `false` is portable across databases?

Reply:
Maybe `1 = 2` is a better choice?

Reply:
Maybe yes, in anycase this statement confuses. It deserves a comment.

Reply:
I will add a comment and change to `1 = 2`

Reply:
BTW, such a "static" string builder gives you zero benefit compared over `+=`. You can also use `+=` and the compliler will transform to string builder automatically. String builders make sense when you have dynamic data like loops.

Reply:
I know. However, the contributing guideline recommends to _copy_ the coding style of the modified source file. That's what I did: the _roles_ and _credentials_ statements have already been there, using the `StringBuilder`.